src/cordova/test.action.mjs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from 'url';
+import os from 'os';
+import minimist from 'minimist';
+import path from 'path';
+import fs from 'fs/promises';
+import rmfr from 'rmfr';
+
+import {getIosDeploymentTarget} from '../build/get_ios_deployment_target.mjs';
+import {getRootDir} from '../build/get_root_dir.mjs';
+import {spawnStream} from '../build/spawn_stream.mjs';
+
+const APPLE_ROOT = path.join(getRootDir(), 'src', 'cordova', 'apple');
+const APPLE_LIBRARY_NAME = 'OutlineAppleLib';
+
+/**
+ * @description Tests the parameterized cordova binary (ios, macos).
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {
+    _: [outlinePlatform],
+  } = minimist(parameters);
+
+  if (outlinePlatform !== 'macos' && outlinePlatform !== 'ios') {
+    throw new Error('Testing is only currently supported for Apple platforms.');
+  }
+
+  if (os.platform() !== 'darwin') {
+    throw new Error('Building an Apple binary requires xcodebuild and can only be done on MacOS');
+  }
+
+  const derivedDataPath = path.join(APPLE_ROOT, 'coverage');
+
+  await rmfr(derivedDataPath);
+  await spawnStream(
+    'xcodebuild',
+    'clean',
+    'test',
+    '-scheme',
+    APPLE_LIBRARY_NAME,
+    '-destination',
+    outlinePlatform === 'macos'
+      ? `platform=macOS,arch=${os.machine()}`
+      : `platform=iOS Simulator,name=${await getIosDeploymentTarget()},OS=latest`,
```

Comment:
The destination is in the wrong format:
![image](https://user-images.githubusercontent.com/113565/222832181-6fd9c9df-3288-4462-b3d5-0540cba72dce.png)

The deployment target is the OS version, not the iPhone number. I think this needs to be
`platform=iOS Simulator,name=iPhone 8,OS=${await getIosDeploymentTarget()}`
or perhaps
`platform=iOS Simulator,name=Any iOS Device,OS=${await getIosDeploymentTarget()}`

Reply:
Yeah, I think just hardcoding is probably the easiest here.